scene/2d/tile_map.h

```diff
@@ -35,403 +35,14 @@
 #include "scene/gui/control.h"
```

Comment:
Unrelated, but this can be forward-declared.

Reply:
Forward declaring in headers helps reduce the amount of files that need to be rebuilt on incremental builds. If a commit changes `control.h`, `tile_map.h` and all the files that include it would need to be recompiled.

If it's forward declared, only `tile_map.cpp` needs to be recompiled.

Reply:
Oh ok, that makes sense! I'll update it then, thanks.